python/huggingfaceserver/huggingfaceserver/model.py

```diff
@@ -73,7 +70,11 @@ def __init__(self, model_name: str, kwargs,
         self.tokenizer = None
         self.model = None
         self.mapping = None
-        self.engine = AsyncLLMEngine.from_engine_args(engine_args) if _vllm else None
+        self.vllm_engine = None
+        self.use_vllm = kwargs.get('use_vllm', True)
+        if self.use_vllm and self.device == torch.device("cuda"):   # vllm needs gpu
+            if self.infer_vllm_supported_from_model_architecture(self.model_id) is not None:
+                self.vllm_engine = AsyncLLMEngine.from_engine_args(engine_args)
```

Comment:
This can be moved to the load handler?

Reply:
Yes, I have made this change in latest commit